rust/datafusion/src/execution/physical_plan/common.rs

```diff
@@ -52,24 +53,21 @@ impl RecordBatchReader for RecordBatchIterator {
         self.schema.clone()
     }
 
-    fn next_batch(&mut self) -> ArrowResult<Option<RecordBatch>> {
-        if self.index < self.batches.len() {
```

Comment:
Yeah, I think the old code here (with `&mut self`) is clearer about what is going on